src/fastfeedparser/main.py

```diff
@@ -44,640 +47,573 @@ def parse(source):
-            entry["author"] = author
-
-        if feed_type == "rss":
-            comments = get_element_value(item, "comments", namespaces)
-            if comments:
-                entry["comments"] = comments
-
-        return entry
+        or get_field_value(
+            "{http://purl.org/dc/elements/1.1/}creator",
+            "{http://purl.org/dc/elements/1.1/}creator",
+            "{http://purl.org/dc/elements/1.1/}creator",
+            False,
+        )
+        or _get_element_value(item, "{http://purl.org/dc/elements/1.1/}creator")
+        or _get_element_value(item, "author")
+    )
+    if author:
+        entry["author"] = author
+
+    if feed_type == "rss":
+        comments = _get_element_value(item, "comments")
+        if comments:
+            entry["comments"] = comments
+
+    return entry
+
+
+def _field_value_getter(
+    root: _Element, feed_type: _FeedType
+) -> Callable[[str, str, str, bool], str | None]:
+    if feed_type == "rss":
+
+        def wrapper(
+            rss_css: str, atom_css: str, rdf_css: str, is_attr: bool
+        ) -> str | None:
+            return _get_element_value(root, rss_css) or (
+                (
+                    _get_element_value(root, atom_css, attribute="href")
+                    or _get_element_value(root, atom_css, attribute="link")
+                )
+                if is_attr
+                else _get_element_value(root, atom_css)
+            )
 
-    # Usage:
-    for item in items:
-        entry = parse_feed_entry(item, feed_type, namespaces)
-        entries.append(entry)
+    elif feed_type == "atom":
 
-    # Trim titles and descriptions
-    for entry in entries:
-        entry["title"] = trim_text(entry.get("title", ""))
-        entry["description"] = trim_text(entry.get("description", ""))
-        entry["summary"] = trim_text(entry.get("summary", ""))
+        def wrapper(
+            rss_css: str, atom_css: str, rdf_css: str, is_attr: bool
+        ) -> str | None:
+            return _get_element_value(root, atom_css) or (
+                (
+                    _get_element_value(root, atom_css, attribute="href")
+                    or _get_element_value(root, atom_css, attribute="link")
+                )
+                if is_attr
+                else None
+            )
 
-    feed["entries"] = entries
-    return feed
+    elif feed_type == "rdf":
 
+        def wrapper(
+            rss_css: str, atom_css: str, rdf_css: str, is_attr: bool
+        ) -> str | None:
+            return _get_element_value(root, rdf_css)
 
-def trim_text(text):
-    """Trim leading and trailing whitespace from text."""
-    return text.strip() if text else ""
+    return wrapper
 
 
-def get_element_value(element, tag, namespaces, attribute=None):
+def _get_element_value(
+    root: _Element, path: str, attribute: str | None = None
+) -> str | None:
     """Get text content or attribute value of an element."""
-    if ":" in tag and not tag.startswith("{"):
```

Comment:
":" in tag is not being checked because I ensured incoming paths contain necessary NS information so it doesn't need to be resolved. Also, resolving ns by their common name is not a good practice because they can be named whatever, only URLs guarantee proper schema.